engine/src/main/java/org/terasology/world/generation/BaseFacetedWorldGenerator.java

```diff
@@ -83,6 +84,15 @@ public WorldConfigurator getConfigurator() {
         return configurator;
     }
 
+    @Override
+    public void setWorldConfigurator(WorldConfigurator configurator) {
+        //TODO: figure out what to do with other world configurator classes...
+        //      maybe this should not be exposed on WorldGenerator at all...
+        if (configurator instanceof FacetedWorldConfigurator) {
+            this.configurator = (FacetedWorldConfigurator) configurator;
+        }
```

Comment:
should this log a warning? (at least until something else is figured out)

Reply:
If we keep that in, probably yes... although I don't like interfaces that depend on being implemented using shabby type casts... :confused: I still hope that we'll find a better way to do this with a better understanding of what is happening here...